pandas/core/internals/blocks.py

```diff
@@ -920,10 +920,11 @@ def setitem(self, indexer, value):
 
             elif lib.is_scalar(value) and not isna(value):
                 dtype, _ = infer_dtype_from_scalar(value, pandas_dtype=True)
-
+            elif is_list_like(value) and is_empty_indexer(indexer, np.array(value)):
```

Comment:
any way to avoid the np.array call?  i think that makes a copy if we have eg a list

Reply:
I don't think so, ``is_empty_indexer`` requires an ndarray, while may be a list.

Reply:
the test is for `loc.__setitem__`.  is that (or i guess `iloc.__setitem__`) the only way we can get here?  it _may_ make sense to do this in setitem_with_indexer

Reply:
Yeah, that would be possible I think.
Sport would be in https://github.com/pandas-dev/pandas/blob/524fc9c6790ddc4b151b6fb910a27073961c5fba/pandas/core/indexing.py#L1567

Ran the tests in indexing, frame/indexing and series/indexing locally, this seems to work. Let me know, if we should move this

Reply:
you can use np.asarray, also can we check len of the indexer here first to short-cut?

Reply:
Thx used asarray. Indexer is a tuple containing an empty list, len does not work here unfortunately.